lxpanel-plugin/kano_updater.c

```diff
@@ -380,71 +382,79 @@ void install_clicked(GtkWidget *widget, gpointer data)
     launch_cmd(SOUND_CMD, NULL);
 }
 
-void check_for_updates_clicked(GtkWidget *widget,
-			       kano_updater_plugin_t *plugin_data)
+void check_for_updates_clicked(GtkWidget *widget, kano_updater_plugin_t *plugin_data)
 {
-	launch_cmd(CHECK_FOR_UPDATES_CMD,  "kano-updater");
+    // launch_cmd can't be used because we need to wait for 
+    // the command to finish
+    kdesk_hourglass_start("kano-dialog");
+    system(CHECK_FOR_UPDATES_CMD);
```

Comment:
@tombettany a bit sceptic of using threads in this case. The UI frozen is the menu, not big deal. If you could test it in today's image and let me know what you think that would help